MotionPlanningGoal/goalComposition.py

```diff
@@ -1,28 +1,49 @@
 from abc import ABC, abstractmethod
+from MotionPlanningGoal.staticSubGoal import StaticSubGoalConfig
 from MotionPlanningSceneHelpers.motionPlanningComponent import MotionPlanningComponent
 from MotionPlanningGoal.subGoalCreator import SubGoalCreator
 from MotionPlanningGoal.staticJointSpaceSubGoal import JointSpaceGoalsNotSupportedError
 
+from dataclasses import dataclass
+from omegaconf import OmegaConf
+from typing import List, Optional, Dict
 
 class MultiplePrimeGoalsError(Exception):
     pass
 
+@dataclass
+class GoalCompositionConfig:
+    """Configuration dataclass for goal composition.
+
+    This configuration class holds all the sub goals 
+    in one dictionary.
+
+    Parameters:
+    ------------
+
+    sub_goal: dict : Map containing all sub goals
+    """
+    sub_goals: Dict[str, StaticSubGoalConfig]
```

Comment:
Good stuff! Just one question, why have you added the `sub_goals` layer here, is it not possible to simply say `OmegaConf.structured(Dict[str, StaticSubGoalConfig])`. Then it won't break the compatibility